internal/backend/db/dbgorm/scheme/scheme.go

```diff
@@ -54,6 +54,10 @@ var Tables = []any{
 
 func unmarshalData(dst any, j datatypes.JSON, c []byte) error {
 	if len(j) == 0 {
+		if len(c) == 0 {
+			return nil
+		}
+
 		r, err := gzip.NewReader(bytes.NewReader(c))
```

Comment:
This is error prone, people need to pay attention to the scope of `r`.

Maybe something like:

```go
func unmarshalData(dst any, j datatypes.JSON, c []byte) error {
	if len(j) == 0 && len(c) == 0 {
		return nil
	}

	if len(j) > 0 && len(c) > 0 {
		return fmt.Errorf("got both compressed and uncompressed data")
	}

	var r io.Reader
	if len(c) > 0 {
		var err error
		r, err = gzip.NewReader(c)
		if err != nil {
			return err
		}
	} else {
		r = bytes.NewReader(j)
	}

	return json.NewDecoder(r).Decode(dst)
}
```